test/e2e/operator_test.go

```diff
@@ -56,6 +64,232 @@ func TestDefaultTunedExists(t *testing.T) {
+	out, err = exec.Command("oc", "apply", "-n", ntoconfig.OperatorNamespace(), "-f", profileHugepages).CombinedOutput()
+	if err != nil {
+		t.Fatal(fmt.Errorf("%v", string(out)))
+	}
+
+	t.Logf("Ensuring the custom worker node profile was set")
+	err = ensureSysctl(sysctlVar, pod, "16")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	t.Logf("Removing label %s from node %s", nodeLabelHugepages, node.Name)
+	out, err = exec.Command("oc", "label", "node", "--overwrite", "-n", ntoconfig.OperatorNamespace(), node.Name, nodeLabelHugepages+"-").CombinedOutput()
+	if err != nil {
+		t.Fatal(fmt.Errorf("%v", string(out)))
+	}
+
+	t.Logf("Ensuring the original %s value (%s) is set in pod %s", sysctlVar, valOrig, pod.Name)
+	err = ensureSysctl(sysctlVar, pod, valOrig)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+// Returns a list of nodes that match a given role.
+func getNodesByRole(cl *client.Client, role string) ([]corev1.Node, error) {
+	nodeList := &corev1.NodeList{}
+	err := (*cl).List(context.TODO(), nodeList, client.MatchingLabels(labels.Set{fmt.Sprintf("node-role.kubernetes.io/%s", role): ""}))
+	if err != nil {
+		return nil, fmt.Errorf("Couldn't get a list of nodes by role (%s): %v", role, err)
+	}
+	return nodeList.Items, nil
+}
+
+// Returns a pod that runs on a given node.
+func getTunedForNode(cl *client.Client, node *corev1.Node) (*corev1.Pod, error) {
+	podList := &corev1.PodList{}
+	err := (*cl).List(context.TODO(), podList, client.InNamespace(ntoconfig.OperatorNamespace()),
+		client.MatchingLabels(labels.Set{"openshift-app": "tuned"}),
+		client.MatchingField("spec.nodeName", node.Name))
+	if err != nil {
+		return nil, fmt.Errorf("Couldn't get a list of tuned pods: %v", err)
+	}
+
+	if len(podList.Items) != 1 {
+		if len(podList.Items) == 0 {
+			return nil, fmt.Errorf("Failed to find a tuned pod for node %s", node.Name)
+		}
+		return nil, fmt.Errorf("Too many (%d) tuned pods for node %s", len(podList.Items), node.Name)
+	}
+	return &podList.Items[0], nil
+}
+
+// Returns a sysctl value for sysctlVar from inside a (tuned) pod.
+func getSysctl(sysctlVar string, pod *corev1.Pod) (val string, err error) {
+	wait.PollImmediate(5*time.Second, 5*time.Minute, func() (bool, error) {
+		var out []byte
+		out, err = exec.Command("oc", "rsh", "-n", ntoconfig.OperatorNamespace(), pod.Name,
+			"sysctl", "-n", sysctlVar).CombinedOutput()
+		if err != nil {
+			// Failed to query a sysctl "sysctlVar" on pod.Name
+			return false, nil
+		}
+		val = strings.TrimSpace(string(out))
+		return true, nil
+	})
+	if err != nil {
+		return "", fmt.Errorf("Failed to retrieve sysctl value %s in pod %s: %v", sysctlVar, pod.Name, err)
+	}
+
+	return val, nil
+}
+
+// Makes sure a sysctl value for sysctlVar from inside a (tuned) pod is equal to valExp.
+// Returns an error otherwise.
+func ensureSysctl(sysctlVar string, pod *corev1.Pod, valExp string) (err error) {
+	var val string
+	wait.PollImmediate(5*time.Second, 5*time.Minute, func() (bool, error) {
```

Comment:
Not sure if this `PollImmediate()` is necessary since `getSysctl()` always has its own?

Reply:
It is not necessary.  I was considering not using it, but then I went for the same pattern and more flexibility (which is probably also questionable).  Would you prefer I removed it in ensureSysctl()?